plugins/glustershd/rest.go

```diff
@@ -201,3 +203,95 @@ func selfHealHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
+
+func splitBrainOperationHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	volname := mux.Vars(r)["volname"]
+	operation := mux.Vars(r)["operation"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req glustershdapi.SplitBrainReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	// check if hostname is in form of uuid, if yes, convert uuid to respective ip addr
+	if uuid.Parse(req.HostName) != nil {
+		if peers, err := peer.GetPeer(req.HostName); err == nil {
+			req.HostName = strings.Split(peers.PeerAddresses[0], ":")[0]
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "couldn't find ip for the peerid")
+			return
+		}
+	}
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	// Validate volume type
+	if !isVolReplicate(volinfo.Type) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid operation for this volume type")
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrVolNotStarted)
+		return
+	}
+
+	var options []string
+	glusterdSockpath := path.Join(config.GetString("rundir"), "glusterd2.socket")
+	switch operation {
+	case "latest-mtime":
+		fallthrough
+	case "bigger-file":
+		if req.FileName != "" {
+			if !strings.HasPrefix(req.FileName, "/") {
+				restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "filename should be an absolute path in volume, should start with / notation")
+				return
+			}
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "please provide filename")
+			return
+		}
+		options = append(options, operation, req.FileName, "glusterd-sock", glusterdSockpath)
+
+	case "source-brick":
+		if req.HostName == "" || req.BrickName == "" {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "please provide hostname and brickname")
+			return
+		}
+		if !volume.CheckBrickExistence(volinfo, req.HostName, req.BrickName) {
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "brick doesn't exist")
```

Comment:
Can we update this message as well mentioning about the host "brick doesn't exist in the provided host"? Something that doesn't confuse the user whether the brick is not in the host. or the brick or the host provided itself is wrong would be better. This might need some update on the CheckBrickExistence function. Being a minor change will enhance the user experience.